thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AccessControlProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package ai.startree.thirdeye.auth;
+
+import ai.startree.thirdeye.spi.accessControl.AccessControl;
+import ai.startree.thirdeye.spi.accessControl.AccessControlConfiguration;
+import ai.startree.thirdeye.spi.accessControl.AccessType;
+import ai.startree.thirdeye.spi.accessControl.ResourceIdentifier;
+
+/**
+ * AccessControlProvider serves as a mutable layer between Guice bindings and the access control implementation from plugins.
+ */
+public class AccessControlProvider implements AccessControl {
+
+  public final static AccessControl alwaysAllow = (
+      final String token,
+      final ResourceIdentifier identifiers,
+      final AccessType accessType
+  ) -> true;
+
+  public final static AccessControl alwaysDeny = (
+      final String token,
+      final ResourceIdentifier identifiers,
+      final AccessType accessType
+  ) -> false;
+
+  private final AccessControlConfiguration config;
+  private AccessControl accessControl = null;
```

Comment:
nit: (For Later) `AccessControl` object is actually the class that checks/enforces the policy. The name is somehow not conveying this. don't have a great suggestion for this. I was thinking `AccessControlEnforcer` or `AccessControlPolicyChecker` or whatever :|